polyfactory/factories/msgspec_factory.py

```diff
@@ -48,6 +50,7 @@ def is_supported_type(cls, value: Any) -> TypeGuard[type[T]]:
         return isclass(value) and hasattr(value, "__struct_fields__")
 
     @classmethod
+    @copying_lru_cache
```

Comment:
I was thinking that maybe we should keep a mapping between the types and the model fields instead of using the decorator. Since we usually create a new factory each time `build` is called, this will end up creating a new cache each time so we won't really get the benefit of the caching right?